Wikimate.php

```diff
@@ -858,3 +858,501 @@ private function findSection($section)
 		return -1;
 	}
 }
+
+
+/**
+ * Models a wiki file that can have its properties retrieved and
+ * its contents downloaded and uploaded.
+ * All properties pertain to the current revision of the file.
+ *
+ * @author  Robert McLeod & Frans P. de Vries
+ * @since   October 2016
+ */
+class WikiFile
+{
+	protected $filename  = null;
+	protected $wikimate  = null;
+	protected $exists    = false;
+	protected $invalid   = false;
+	protected $error     = null;
+	protected $edittoken = null;
+	protected $info      = null;
+
+	/*
+	 *
+	 * Magic methods
+	 *
+	 */
+
+	/**
+	 * Constructs a WikiFile object from the filename given
+	 * and adds a Wikimate object.
+	 *
+	 * @param  string    $filename  Name of the wiki file
+	 * @param  Wikimate  $wikimate  Wikimate object
+	 */
+	public function __construct($filename, $wikimate)
+	{
+		$this->wikimate = $wikimate;
+		$this->filename = $filename;
+		$this->info     = $this->getInfo(true);
+
+		if ($this->invalid) {
+			$this->error['file'] = 'Invalid filename - cannot create WikiFile';
+		}
+	}
+
+	/**
+	 * Forget all object properties.
+	 *
+	 * @return  <type>  Destructor
+	 */
+	public function __destruct()
+	{
+		$this->filename   = null;
+		$this->wikimate   = null;
+		$this->exists     = false;
+		$this->invalid    = false;
+		$this->error      = null;
+		$this->edittoken  = null;
+		$this->info       = null;
+		return null;
+	}
+
+	/**
+	 * Returns the file existance status.
+	 *
+	 * @return  boolean  True if file exists
+	 */
+	public function exists()
+	{
+		return $this->exists;
+	}
+
+	/**
+	 * Alias of self::__destruct().
+	 */
+	public function destroy()
+	{
+		$this->__destruct();
+	}
+
+	/*
+	 *
+	 * File meta methods
+	 *
+	 */
+
+	/**
+	 * Returns an error if there is one, null shows no error.
+	 *
+	 * @return  mixed  Null for no errors, or an error array
+	 */
+	public function getError()
+	{
+		return $this->error;
+	}
+
+	/**
+	 * Returns the name of this file.
+	 *
+	 * @return  string  The name of this file
+	 */
+	public function getFilename()
```

Comment:
Like `getTitle` in WikiPage, this method returns the input parameter, not a file property (`getCanonicalTitle` does that), hence I'm keeping them in the "meta methods" section.